DiscordChatExporter.Core/Exporting/Filtering/ContainsMessageFilter.cs

```diff
@@ -0,0 +1,16 @@
+﻿using DiscordChatExporter.Core.Discord.Data;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public class ContainsMessageFilter : MessageFilter
+    {
+        private readonly string _value;
+
+        public ContainsMessageFilter(string value)
+        {
+            _value = value;
+        }
+
+        public override bool Filter(Message message) => message.Content.Contains(_value);
```

Comment:
This is a really primitive check that should probably be updated. Right now the token "a" will match any word with a lowercase letter a (e.g., it'll match the message "banana" but not the message "A"). We could try splitting the value and message content into words and making sure all words in the filter value are also in the message.

Reply:
Just an idea, but maybe we should make it so the contains filter tries to match a word? So `a` would only match standalone `a` word (the article) separate by whitespace, but not as part of another word. Is that how Discord search works too, I wonder? In any case, this can be achieved by using `Regex.IsMatch(message.Content, "\\b" + Regex.Esacpe(_value) + "\\b")`

Reply:
Yeah, I think that's better. Discord search seems to mostly match the same way but also attempts to finds closely related words with some strange results ("AC" matches "ace", "youings" matches "you", etc.). Should we make this case-insensitive?

Reply:
Yeah let's make it case-insensitive but not bother with fuzzy matching (like ace, etc) for now